src/assembler/parser/text.ts

```diff
@@ -0,0 +1,73 @@
+/**
+ * Represents a text cursor which points to a specific location in a text.
+ */
+export interface ITextCursor {
+  index: number
+  line: number
+  position: number
+}
+
+/**
+ * Represents a text match during parsing.
+ */
+export interface ITextMatch {
+  from: ITextCursor
+  to: ITextCursor
+  text: string
+  captures: string[]
+}
+
+/**
+ * Represents a rule for parsing text.
+ */
+export interface ITextParseRule {
+  name: string
+  pattern: RegExp
+  onMatch?(match: ITextMatch): void
+}
+
+/**
+ * Parses text using the specified rules.
+ *
+ * @param text the text to be parsed
+ * @param rules the rules to be used for parsing
+ * @returns the last cursor position before there was no more match
+ */
+export function parseText(text: string, rules: ITextParseRule[]): ITextCursor {
```

Comment:
Maybe it would make sense to also add dedicated tests for this module? But perhaps it is already more than enough with covered with the parser tests.

Reply:
Inside of the parser tests, there are also tests for the text parser. I've created one test file for the whole parser folder.